src/ring_theory/witt_vector/structure_polynomial.lean

```diff
@@ -317,4 +326,231 @@ begin
+end
+
+@[simp] lemma witt_sub_zero : witt_sub p 0 = X (0,0) - X (1,0) :=
+begin
+  apply mv_polynomial.map_injective (int.cast_ring_hom ℚ) int.cast_injective,
+  simp only [witt_sub, witt_structure_rat, alg_hom.map_sub, ring_hom.map_sub,
+    rename_X, X_in_terms_of_W_zero, map_X,
+     witt_polynomial_zero, bind₁_X_right, map_witt_structure_int],
+end
+
+@[simp] lemma witt_mul_zero : witt_mul p 0 = X (0,0) * X (1,0) :=
+begin
+  apply mv_polynomial.map_injective (int.cast_ring_hom ℚ) int.cast_injective,
+  simp only [witt_mul, witt_structure_rat, rename_X, X_in_terms_of_W_zero, map_X,
+    witt_polynomial_zero, ring_hom.map_mul,
+    bind₁_X_right, alg_hom.map_mul, map_witt_structure_int]
+
+end
+
+@[simp] lemma witt_neg_zero : witt_neg p 0 = - X (0,0) :=
+begin
+  apply mv_polynomial.map_injective (int.cast_ring_hom ℚ) int.cast_injective,
+  simp only [witt_neg, witt_structure_rat, rename_X, X_in_terms_of_W_zero, map_X,
+    witt_polynomial_zero, ring_hom.map_neg,
+   alg_hom.map_neg, bind₁_X_right, map_witt_structure_int]
+end
+
+@[simp] lemma constant_coeff_witt_add (n : ℕ) :
+  constant_coeff (witt_add p n) = 0 :=
+begin
+  apply constant_coeff_witt_structure_int p _ _ n,
+  simp only [add_zero, ring_hom.map_add, constant_coeff_X],
+end
+
+@[simp] lemma constant_coeff_witt_sub (n : ℕ) :
+  constant_coeff (witt_sub p n) = 0 :=
+begin
+  apply constant_coeff_witt_structure_int p _ _ n,
+  simp only [sub_zero, ring_hom.map_sub, constant_coeff_X],
+end
+
+@[simp] lemma constant_coeff_witt_mul (n : ℕ) :
+  constant_coeff (witt_mul p n) = 0 :=
+begin
+  apply constant_coeff_witt_structure_int p _ _ n,
+  simp only [mul_zero, ring_hom.map_mul, constant_coeff_X],
+end
+
+@[simp] lemma constant_coeff_witt_neg (n : ℕ) :
+  constant_coeff (witt_neg p n) = 0 :=
+begin
+  apply constant_coeff_witt_structure_int p _ _ n,
+  simp only [neg_zero, ring_hom.map_neg, constant_coeff_X],
+end
+
+end witt_structure_simplifications
+
+section witt_vars
+variable (R)
+
+-- we could relax the fintype on `idx`, but then we need to cast from finset to set.
+-- for our applications `idx` is always finite.
+lemma witt_structure_rat_vars [fintype idx] (Φ : mv_polynomial idx ℚ) (n : ℕ) :
+  (witt_structure_rat p Φ n).vars ⊆ finset.univ.product (finset.range (n + 1)) :=
+begin
+  rw witt_structure_rat,
+  intros x hx,
+  simp only [finset.mem_product, true_and, finset.mem_univ, finset.mem_range],
+  replace hx := vars_bind₁ _ _ hx,
+  simp only [exists_prop, finset.mem_bind, finset.mem_range] at hx,
+  rcases hx with ⟨k, hk, hx⟩,
+  replace hk := X_in_terms_of_W_vars_subset p _ hk,
+  rw finset.mem_range at hk,
+  replace hx := vars_bind₁ _ _ hx,
+  simp only [exists_prop, finset.mem_bind, finset.mem_range] at hx,
+  rcases hx with ⟨i, -, hx⟩,
+  replace hx := vars_rename _ _ hx,
+  rw [finset.mem_image] at hx,
+  rcases hx with ⟨j, hj, rfl⟩,
+  rw [witt_polynomial_vars, finset.mem_range] at hj,
+  exact lt_of_lt_of_le hj hk,
+end
```

Comment:
This proof is a bit hard to read, but the only way I can think of to improve it is to add a `mem_vars_$foo` lemma that combines the `replace hx := vars_$foo hx, simp at hx` steps. For example, we could write `obtain ⟨k, hk, hx⟩ := mem_vars_bind₁ _ _ hx` instead of the `replace hx := _, simp at hx, rcases hx` we have now.

Reply:
Good idea, I will work on that lemma asap. Thanks for the review!

Reply:
Done